src/coreComponents/physicsSolvers/wavePropagation/AcousticVTIWaveEquationSEM.cpp

```diff
@@ -0,0 +1,871 @@
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOS Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * @file AcousticVTIWaveEquationSEM.cpp
+ */
+
+#include "AcousticVTIWaveEquationSEM.hpp"
+#include "AcousticVTIWaveEquationSEMKernel.hpp"
+
+#include "finiteElement/FiniteElementDiscretization.hpp"
+#include "fieldSpecification/FieldSpecificationManager.hpp"
+#include "fieldSpecification/PerfectlyMatchedLayer.hpp"
+#include "mainInterface/ProblemManager.hpp"
+#include "mesh/ElementType.hpp"
+#include "mesh/mpiCommunications/CommunicationTools.hpp"
+#include "WaveSolverUtils.hpp"
+
+namespace geos
+{
+
+using namespace dataRepository;
+
+AcousticVTIWaveEquationSEM::AcousticVTIWaveEquationSEM( const std::string & name,
+                                                  Group * const parent ):
+  WaveSolverBase( name,
+                  parent )
+{
+
+  registerWrapper( viewKeyStruct::sourceNodeIdsString(), &m_sourceNodeIds ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Indices of the nodes (in the right order) for each source point" );
+
+  registerWrapper( viewKeyStruct::sourceConstantsString(), &m_sourceConstants ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Constant part of the source for the nodes listed in m_sourceNodeIds" );
+
+  registerWrapper( viewKeyStruct::sourceIsAccessibleString(), &m_sourceIsAccessible ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Flag that indicates whether the source is accessible to this MPI rank" );
+
+  registerWrapper( viewKeyStruct::receiverNodeIdsString(), &m_receiverNodeIds ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Indices of the nodes (in the right order) for each receiver point" );
+
+  registerWrapper( viewKeyStruct::receiverConstantsString(), &m_receiverConstants ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Constant part of the receiver for the nodes listed in m_receiverNodeIds" );
+
+  registerWrapper( viewKeyStruct::receiverIsLocalString(), &m_receiverIsLocal ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Flag that indicates whether the receiver is local to this MPI rank" );
+
+  registerWrapper( viewKeyStruct::pressureNp1AtReceiversString(), &m_pressureNp1AtReceivers ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Pressure value at each receiver for each timestep" );
+
+}
+
+AcousticVTIWaveEquationSEM::~AcousticVTIWaveEquationSEM()
+{
+  // TODO Auto-generated destructor stub
```

Comment:
Is this destructor needed?

Reply:
Probably not because it is defined (empty) in the mother class `WaveSolverBase`. It also exists in the acoustic propagator in `AcousticWaveEquationSEM.cpp`

```
AcousticWaveEquationSEM::~AcousticWaveEquationSEM()
{
  // TODO Auto-generated destructor stub
}
```

Let me remove it and try :)